### PR TITLE
Add GitHub Actions CI to test the build

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This will hopefully ensure that future pull requests get basic compilation testing to prevent broken merges.

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
I don't know what's going on here, but it definitely runs in my branch: https://github.com/Conan-Kudo/wf-recorder/actions/runs/3771840054/jobs/6412444975

**Comment by Conan-Kudo** [comment]
It's probably not showing up on the pull request here because actions might not be enabled in this repo yet?

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ sudo apt install libavutil-dev libavcodec-dev libavformat-dev libswscale-dev lib
 
 #### Fedora
 ```
-$ sudo dnf install wayland-devel wayland-protocols-devel ffmpeg-free-devel
+$ sudo dnf install wayland-devel wayland-protocols-devel ffmpeg-free-devel pulseaudio-libs-devel
```
Comment (ammen99) [inline README.md:57]: Seems like this doesn't belong in this commit.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+              'pkgconfig(wayland-client)' 'pkgconfig(wayland-protocols)' 'pkgconfig(libpulse-simple)'
+              'pkgconfig(libavutil)' 'pkgconfig(libavcodec)' 'pkgconfig(libavformat)'
+              'pkgconfig(libavdevice)' 'pkgconfig(libavfilter)' 'pkgconfig(libswresample)'
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones speed things up
+      - name: meson configure
+        run: meson ./Build
+      - name: compile with ninja
+        run: ninja -C ./Build -j4
```
Comment (ammen99) [inline .github/workflows/build.yaml:30]: Does GitHub have more than 1 build job? ninja usually should autodetect the best amount of threads to use.

Comment (Conan-Kudo) [reply .github/workflows/build.yaml:30]: I usually set it to a specific number of cores to make it predictable, but I can drop it to make it dynamic...

Comment (ammen99) [reply .github/workflows/build.yaml:30]: I think hardcoding 4 threads doesn't make sense for a GitHub Action. AFAIK they are single threaded anyway ..

File: README.md (hunk 1 above)
Comment (Conan-Kudo) [reply README.md]: Dropped and moved to #203

File: .github/workflows/build.yaml (hunk 2 above)
Comment (Conan-Kudo) [reply .github/workflows/build.yaml]: Dropped.

**Comment by ammen99** [comment]
Let's merge and see whether the Github Action will be enabled automatically (I think it should be). In any case, thanks, this is a very useful addition :)
